server/api/src/rsGenQuery.cpp

```diff
@@ -414,30 +421,23 @@ irods::error strip_resc_hier_name_from_query_inp( genQueryInp_t* _inp, int& _pos
     // fresh indices and values
     std::memset(&_inp->sqlCondInp, 0, sizeof(_inp->selectInp));
 
-    // =-=-=-=-=-=-=-
-    // iterate over tmp and replace resource group with resource name
-    for ( int i = 0; i < tmpV.len; ++i ) {
-        if ( tmpV.inx[i] == COL_D_RESC_HIER ) {
-            std::string new_cond = "='0'";
-            irods::error ret = get_resc_id_cond_for_hier_cond(
-                                   tmpV.value[i],
-                                   new_cond);
-            if(!ret.ok()) {
-                irods::log(PASS(ret));
+    try {
+        // iterate over tmp and replace resource group with resource name
+        for (int i = 0; i < tmpV.len; ++i) {
+            if (tmpV.inx[i] == COL_D_RESC_HIER) {
+                const auto new_cond = get_resc_id_cond_for_hier_cond(tmpV.value[i]);
+                addInxVal(&_inp->sqlCondInp, COL_D_RESC_ID, new_cond.empty() ? "='0'" : new_cond.data());
```

Comment:
We should start using `.c_str()` instead of `.data()`.

`.c_str()` implies a null-terminated string. Not only that, but based on the following blog post, using `.c_str()` protects us from things like `new_cond` becoming a `std::string_view` in the future.
- https://quuxplusone.github.io/blog/2020/03/20/c-str-correctness/#data-versus-c_str

Reply:
Thanks, will do